ndb/src/google/cloud/ndb/model.py

```diff
@@ -1833,17 +1833,122 @@ def _db_get_value(self, v, unused_p):
 
 
 class TextProperty(BlobProperty):
+    """An unindexed property that contains UTF-8 encoded text values.
+
+    A :class:`TextProperty` is intended for values of unlimited length, hence
+    is **not** indexed by default. A :class:`TextProperty` can be indexed via:
+
+    .. code-block:: python
+
+        class Item(ndb.Model):
+            description = ndb.TextProperty(indexed=True)
+            ...
+
+    but that is not the intended usage. If indexed text is desired, a
```

Comment:
@chrisrossi Do you think we should make this explicit by removing `indexed` from the constructor and making

```python
@property
def _indexed(self):
    return False
```

Reply:
Done